rollup.config.js

```diff
@@ -57,36 +57,48 @@ export default [
   mainConfig('system', false),
   mainConfig('s', true),
   mainConfig('s', false),
+  mainConfig('node', true),
+  mainConfig('node', false),
   ...extrasConfig(true),
   ...extrasConfig(false)
 ];
 
 function mainConfig(name, isDev) {
   const sjs = name === 's';
+  const node = name === 'node';
   let banner;
   if (sjs) {
     banner = isDev ? `/*
   * SJS ${version}
   * Minimal SystemJS Build
   */` : null;
+  } else if (node) {
+    banner = `/*
+  * SystemJS ${version}
+  * NodeJS Build
+  */`;
   } else {
     banner = `/*
   * SystemJS ${version}
   */`;
   }
 
+  name = node ? 'system-node' : name;
+  const outputFormat = node ? 'cjs' : 'iife';
+
   return {
     input: `src/${name}.js`,
     output: {
-      file: `dist/${name}${isDev ? '' : '.min'}.js`,
-      format: 'iife',
+      file: `dist/${name}${isDev ? '' : '.min'}.${node ? 'c' : ''}js`,
+      format: outputFormat,
       strict: false,
       sourcemap: !isDev,
       banner
     },
     plugins: [
       replace({
         TRACING: sjs ? 'false' : 'true',
+        [`typeof document `]: node ? `"undefined"` : `"object"`
```

Comment:
This makes me a little uneasy. It's generally not a good idea to make blanket assumptions about environments. Worth remembering SystemJS should work anywhere, including environments that mix semantics such as electron, Chrome extensions, JSDom, etc etc.

The browser build is already based to pretty minimum assumptions - is there really much saving?

Reply:
👍 removed